AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -300,25 +316,33 @@ - (void)deleteDocumentWithPartition:(NSString *)partition
                          documentId:(NSString *)documentId
                        writeOptions:(MSWriteOptions *_Nullable)__unused writeOptions
                   completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
-  [self performOperationForPartition:partition
-                          documentId:documentId
-                          httpMethod:kMSHttpMethodDelete
-                                body:[NSData data]
-                   additionalHeaders:nil
-                   completionHandler:^(NSData *_Nullable __unused responseBody, NSHTTPURLResponse *_Nullable __unused response,
-                                       NSError *_Nullable cosmosDbError) {
-                     // Body returned from call (data) is empty.
-                     NSInteger httpStatusCode = [MSDataSourceError errorCodeFromError:cosmosDbError];
-                     if (httpStatusCode != MSHTTPCodesNo204NoContent) {
-                       MSLogError([MSDataStore logTag],
-                                  @"Not able to delete document. Error: %@; HTTP status code: %ld; "
-                                  @"Document: %@/%@",
-                                  cosmosDbError.localizedDescription, (long)httpStatusCode, partition, documentId);
-                     } else {
-                       MSLogDebug([MSDataStore logTag], @"Document deleted: %@/%@", partition, documentId);
-                     }
-                     completionHandler([[MSDataSourceError alloc] initWithError:cosmosDbError]);
-                   }];
+
+  @synchronized(self) {
+    if (![self canBeUsed] || ![self isEnabled]) {
+      NSError *error = [self generateDisabledError:@"delete" documentId:documentId];
+      completionHandler([[MSDataSourceError alloc] initWithError:error errorCode:MSACDocumentUnknownErrorCode]);
```

Comment:
Why is the code `MSACDocumentUnknownErrorCode`? I wonder why it takes `errorCode` along with `error` instance.

Reply:
I guess my question is why it's not `documentId` like it is for `readWithPartition`

Reply:
@jaelim-ms This errorCode expects to be from an enum that defines only http errors. The NSError has its own code for disabled.

Reply:
> I guess my question is why it's not `documentId` like it is for `readWithPartition`

We are using documentId here too, I don't understand the feedback.

Reply:
Also its different completion handlers for different APIs it's unrelated to our changes.